datalake/backend/endpoints/dataset_versions/create.py

```diff
@@ -0,0 +1,106 @@
+"""Import dataset-version function."""
+import json
+import logging
+import uuid
+from os import environ
+
+import boto3
+from jsonschema import ValidationError, validate  # type: ignore[import]
+from pynamodb.exceptions import DoesNotExist
+
+from ..datasets.model import DatasetModel
+from ..utils import DATASET_TYPES, ENV, JSON_OBJECT, error_response, success_response
+
+stepfunc_client = boto3.client("stepfunctions")
+ssm_client = boto3.client("ssm")
+
+STEP_FUNCTION_ARN_PARAMETER_NAME = f"/{ENV}/StepFuncStateMachineARN"
+
+
+def create_dataset_version(payload: JSON_OBJECT) -> JSON_OBJECT:
+    """POST: Create Dataset."""
+
+    logger = set_up_logging()
+
+    logger.debug(json.dumps({"payload": payload}))
+
+    BODY_SCHEMA = {
+        "type": "object",
+        "properties": {
+            "id": {"type": "string"},
+            "type": {
+                "type": "string",
+                "enum": DATASET_TYPES,
+            },
+            "metadata-url": {"type": "string"},
+        },
+        "required": ["id", "metadata-url", "type"],
+    }
+
+    # validate input
+    req_body = payload["body"]
+    try:
+        validate(req_body, BODY_SCHEMA)
+    except ValidationError as err:
+        logger.warning(json.dumps({"error": err}, default=str))
+        return error_response(400, err.message)
+
+    # validate dataset exists
+    try:
+        dataset = DatasetModel.get(
+            hash_key=f"DATASET#{req_body['id']}",
+            range_key=f"TYPE#{req_body['type']}",
+            consistent_read=True,
+        )
+    except DoesNotExist as err:
+        logger.warning(json.dumps({"error": err}, default=str))
+        return error_response(404, f"dataset '{req_body['id']}' could not be found")
+
+    transaction_id = str(uuid.uuid1())
```

Comment:
I would rename variable `transaction_id` to `dataset_version_id` or `version_id`